engines/shared/app/errors/shared/error.rb

```diff
@@ -0,0 +1,27 @@
+module Shared
+  class Error < Shared::BaseSerializer
```

Comment:
This class acts as a factory and fills the json response with the data the error class provides